bodhi/server/models.py

```diff
@@ -2056,24 +2056,48 @@ def check_requirements(self, session, settings):
             return False, "Failed to determine last_modified: %r" % e.message
 
         try:
-            query = dict(title=self.title, since=since)
+            # query results for this update
+            query = dict(type='bodhi_update', item=self.alias, since=since,
+                         testcases=','.join(requirements))
             results = bodhi.server.util.taskotron_results(settings, **query)
-        except IOError as e:
-            return False, "Failed to talk to taskotron: %r" % e.message
+
+            # query results for each build
+            # retrieve timestamp for each build so that queries can be optimized
+            koji = buildsys.get_session()
+            koji.multicall = True
+            for build in self.builds:
+                koji.getBuild(build.nvr)
+            buildinfos = koji.multiCall()
+
+            for index, build in enumerate(self.builds):
+                buildinfo = buildinfos[index]
+                if not isinstance(buildinfo, dict):
+                    log.warn("Koji build info doesn't seem correct: %r",
+                             buildinfo)
+                ts = datetime.utcfromtimestamp(buildinfo['completion_ts']).isoformat()
+
+                query = dict(type='koji_build', item=build.nvr, since=ts,
+                             testcases=','.join(requirements))
+                build_results = bodhi.server.util.taskotron_results(settings, **query)
+                results.extend(build_results)
+
+        except Exception as e:
+            return False, "Failed retrieving requirements results: %r" % e.message
```

Comment:
We should probably do a log.exception() here too so we can have a record of what happened.